syntax = "proto3";

package envoy.service.secret.v3alpha;

import "envoy/service/discovery/v3alpha/discovery.proto";

import "google/api/annotations.proto";

import "udpa/annotations/versioning.proto";

import "envoy/annotations/resource.proto";

option java_package = "io.envoyproxy.envoy.service.secret.v3alpha";
option java_outer_classname = "SdsProto";
option java_multiple_files = true;
option java_generic_services = true;

service SecretDiscoveryService {
  option (envoy.annotations.resource).type =
      "envoy.extensions.transport_sockets.tls.v3alpha.Secret";

  rpc DeltaSecrets(stream discovery.v3alpha.DeltaDiscoveryRequest)
      returns (stream discovery.v3alpha.DeltaDiscoveryResponse) {
  }

  rpc StreamSecrets(stream discovery.v3alpha.DiscoveryRequest)
      returns (stream discovery.v3alpha.DiscoveryResponse) {
  }

  rpc FetchSecrets(discovery.v3alpha.DiscoveryRequest)
      returns (discovery.v3alpha.DiscoveryResponse) {
    option (google.api.http).post = "/v3alpha/discovery:secrets";
    option (google.api.http).body = "*";
  }
}

// [#not-implemented-hide:] Not configuration. Workaround c++ protobuf issue with importing
// services: https://github.com/google/protobuf/issues/4221
message SdsDummy {
  option (udpa.annotations.versioning).previous_message_type =
      "envoy.service.discovery.v2.SdsDummy";
}
